tests/unit/sagemaker/workflow/test_properties.py

```diff
@@ -92,3 +93,35 @@ def test_properties_describe_model_package_output():
     assert prop.ValidationSpecification.ValidationRole.expr == {
         "Get": "Steps.MyStep.ValidationSpecification.ValidationRole"
     }
+
+
+def test_to_string():
+    prop = Properties("Steps.MyStep", "DescribeTrainingJobResponse")
+
+    assert prop.CreationTime.to_string().expr == {
+        "Std:Join": {
+            "On": "",
+            "Values": [{"Get": "Steps.MyStep.CreationTime"}],
+        },
+    }
+
+    with pytest.raises(TypeError) as error:
+        str(prop.CreationTime)
```

Comment:
I think we should cover all overrides, i.e. str(), float(), int(), as property reference can be resolved to any of them

Reply:
Yup, the `__init__` and `__float__` have been covered in `PipelineVariable` class but are missing in the unit tests. Will add tests to cover.